linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/GenIncOptimizer.scala

```diff
@@ -66,19 +69,19 @@ abstract class GenIncOptimizer private[optimizer] (config: CommonPhaseConfig) {
   private var batchMode: Boolean = false
 
   private var objectClass: Class = _
-  private val classes = CollOps.emptyMap[ClassName, Class]
+  private val classes = new ConcurrentHashMap[ClassName, Class]
```

Comment:
Scala's `mutable.TrieMap` is a fairly specific data structure, which is different from a `ConcurrentHashMap`. Are we sure that this change does not create performance regressions, notably in highly contended cases?

Reply:
I can run some basic benchmarks, but I don't have a beefy machine available (my dev machine only has 4 virtual cores). So it's unlikely we'd discover any serious contention issues.

Reply:
Alternatively, I can just keep `CollOps.Map`. Basically all other cleanups are still doable.

(I am concerned that this sets a precedent of "we are too afraid to change something", which is a dangerous place to be in. But that is a meta-point that allows for longer discussion; my main goal here is to unblock further work on #4315).

Reply:
I'm not opposed to the change. I just noticed that it was a real change and brought it up for consideration. At the end of the day, you are the parallelism expert in that area of the code, so if you think it's no big deal, I'm fine with it.

Reply:
Yeah, but now that you mentioned it, I *do* think we shouldn't just change it :P I'll try to not touch any of the parallelization properties of the code but just clean up a bit, if I'm able to do that, I'll settle for that.